src/filterTreeViewProvider.ts

```diff
@@ -1,68 +1,103 @@
 import * as vscode from "vscode";
-import { Filter } from "./utils";
+import { Filter, Group } from "./utils";
 
 //provides filters as tree items to be displayed on the sidebar
-export class FilterTreeViewProvider
-  implements vscode.TreeDataProvider<FilterItem>
-{
-  constructor(private filterArr: Filter[]) {}
+export class FilterTreeViewProvider implements vscode.TreeDataProvider<vscode.TreeItem> {
+  constructor(private groupArr: Group[]) { }
 
-  getTreeItem(element: FilterItem): vscode.TreeItem {
+  getTreeItem(element: vscode.TreeItem): vscode.TreeItem {
     return element;
   }
 
-  //getChildren(filterItem) returns empty list because filters have no children.
+  //getChildren(vscode.TreeItem) returns empty list because filters have no children.
   //getChildren() returns the root elements (all the filters)
-  getChildren(element?: FilterItem): FilterItem[] {
-    if (element) {
-      return [];
+  getChildren(element?: vscode.TreeItem): Thenable<vscode.TreeItem[]> {
```

Comment:
Why do you choose to use Thenable instead of just returning a vscode.TreeItem[]?

Reply:
I have confirmed that in various official examples [vscode-extension-samples](https://github.com/microsoft/vscode-extension-samples), getChildren is implemented using Thenable and handled asynchronously.

Reply:
yes but the return type can also be non async, and there is no clear reason why needing the async here. As far as I know Promise.resolve() creates a async that is in a sync underneath. 
I don't have a very strong opinion here, so the above is just for discussion here.